dmoj/graders/standard.py

```diff
@@ -101,18 +101,19 @@ def check_result(self, case, result):
         # We shouldn't run checkers if the submission is already known to be incorrect, because some checkers
         # might be very computationally expensive.
         # See https://github.com/DMOJ/judge/issues/170
-        if not result.result_flag:
+        checker = case.checker()
+        if not result.result_flag or getattr(checker.func, 'run_on_error', False):
```

Comment:
Did you test this code?

```python
In [1]: def check(foo): 
   ...:     print(foo) 
   ...:                                                                                                                                                                                                                                                                                                                

In [2]: check.run_on_error = True   

In [4]: getattr(check, "run_on_error", False)                                                                                                                                                                                                                                                                          
Out[4]: True

In [9]: getattr(check.func, "run_on_error", False)                                                                                                                                                                                                                                                                     
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-6966974e4b7e> in <module>
----> 1 getattr(check.func, "run_on_error", False)

AttributeError: 'function' object has no attribute 'func'
```

Reply:
`check` is a function, but `checker` is a `partial` object